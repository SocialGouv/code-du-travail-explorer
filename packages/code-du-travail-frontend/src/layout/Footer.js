import React from "react";
import getConfig from "next/config";
import { Link } from "../../routes";

const { publicRuntimeConfig } = getConfig();

const Footer = () => (
  <footer className="section-dark site-footer" id="footer">
    <div className="container">
      <div className="main-footer">
        <div className="footer__block">
          <h3>Code du travail numérique</h3>
          <ul>
            <li>
              <Link route="about">
                <a>À propos</a>
              </Link>
            </li>
            <li>
              <a href="mailto:contact@code-du-travail.beta.gouv.fr">Contact</a>
            </li>
          </ul>
        </div>
        <div className="footer__block">
          <h3>Aidez-nous à améliorer cet outil</h3>
          <ul>
            <li>
              <a
                href={
                  "https://github.com/SocialGouv/code-du-travail-numerique/tree/v" +
                  publicRuntimeConfig.PACKAGE_VERSION
                }
                className="external-link__after"
              >
                Contribuer sur Github
              </a>
            </li>
            <li>
              <a
                href={
                  "https://github.com/SocialGouv/code-du-travail-numerique/releases/tag/v" +
                  publicRuntimeConfig.PACKAGE_VERSION
                }
                className="external-link__after"
              >
                Journal des modifications
              </a>
            </li>
          </ul>
        </div>
        <div className="footer__block">
          <h3>Liens et outils</h3>
          <ul>
            <li>
              <a
                href="https://socialgouv.github.io/faq-code-du-travail"
                className="external-link__after"
              >
                Question-Réponse des services de renseignement
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </footer>
);

export default Footer;
