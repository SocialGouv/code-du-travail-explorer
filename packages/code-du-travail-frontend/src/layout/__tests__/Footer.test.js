import React from "react";
import { render } from "react-testing-library";
import Footer from "../Footer";

jest.mock(
  "next/config",
  () => () => ({
    publicRuntimeConfig: {
      API_URL: "api.url"
    }
  }),
  { virtual: true }
);

describe("<Footer />", () => {
  it("should render", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
