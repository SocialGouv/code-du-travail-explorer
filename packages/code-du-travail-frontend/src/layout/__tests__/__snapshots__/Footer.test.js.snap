// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 1`] = `
<div>
  <footer
    class="section-dark site-footer"
    id="footer"
  >
    <div
      class="container"
    >
      <div
        class="main-footer"
      >
        <div
          class="footer__block"
        >
          <h3>
            Code du travail numérique
          </h3>
          <ul>
            <li>
              <a
                href="/a-propos"
              >
                À propos
              </a>
            </li>
            <li>
              <a
                href="mailto:contact@code-du-travail.beta.gouv.fr"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
        <div
          class="footer__block"
        >
          <h3>
            Aidez-nous à améliorer cet outil
          </h3>
          <ul>
            <li>
              <a
                class="external-link__after"
                href="https://github.com/SocialGouv/code-du-travail-numerique/tree/vx.y.z"
              >
                Contribuer sur Github
              </a>
            </li>
            <li>
              <a
                class="external-link__after"
                href="https://github.com/SocialGouv/code-du-travail-numerique/releases/tag/vx.y.z"
              >
                Journal des modifications
              </a>
            </li>
          </ul>
        </div>
        <div
          class="footer__block"
        >
          <h3>
            Liens et outils
          </h3>
          <ul>
            <li>
              <a
                class="external-link__after"
                href="https://socialgouv.github.io/faq-code-du-travail"
              >
                Question-Réponse des services de renseignement
              </a>
            </li>
            <li>
              <div
                class="btn btn__link  "
                link="true"
                role="button"
                tabindex="0"
              >
                Trouvez votre convention collective
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
